src/main/kotlin/org/rust/lang/core/resolve/RustResolveEngine.kt

```diff
@@ -140,15 +140,32 @@ private class Resolver {
             if (isPlainPathImport) {
                 val name = path.alias ?: pathPart
                 if (match(name)) {
-                    if (use in visitedImports) {
+                    if (alreadyVisited(use)) {
                         return
                     }
-                    visitedImports += use
-
                     val item = resolve(pathPart).element ?: return
                     return found(item)
                 }
             }
+
+            // `use foo::{self, bar as baz}`
+            if (path.lbrace != null) {
+                for (glob in path.useGlobList) {
+                    if (match(glob.boundElement ?: continue)) {
+                        if (alreadyVisited(glob)) {
+                            return
+                        }
+                        val item = resolveUseGlob(glob).element ?: return
+                        return  found(item)
```

Comment:
:( fixed. 

What is the best strategy to deal with formatting? I can't just reformat the code because it breaks alignment.